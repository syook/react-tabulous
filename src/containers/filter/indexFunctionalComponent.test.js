import React from 'react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import FilterProvider from './indexFunctionalComponent';

configure({ adapter: new Adapter() });

const data = [{ Description: 'Dev', Name: 'David', id: 1, objIndex: 0 }];

describe('FilterProvider', () => {
  it('should render without crashing', () => {
    shallow(<FilterProvider data={[]} />);
  });

  it('should match snapshot if data is empty', () => {
    const tree = shallow(<FilterProvider data={[]} />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should match snapshot if data is not empty', () => {
    const tree = shallow(<FilterProvider data={data} filterableColumns={[]} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
