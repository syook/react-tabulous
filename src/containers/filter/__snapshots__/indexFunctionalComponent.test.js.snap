// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterProvider should match snapshot if data is empty 1`] = `
<ContextProvider
  value={
    Object {
      "count": undefined,
      "data": Array [],
      "filterDisabled": false,
      "rawData": undefined,
      "selectedFilters": Array [],
      "shouldFilterReset": false,
    }
  }
>
  <TableFilter
    applyFilter={[Function]}
    disabled={true}
    filterDisabled={false}
    selectedFilters={Array []}
    setSelectedFilters={[Function]}
    shouldFilterReset={false}
  />
</ContextProvider>
`;

exports[`FilterProvider should match snapshot if data is not empty 1`] = `
<ContextProvider
  value={
    Object {
      "count": undefined,
      "data": Array [
        Object {
          "Description": "Dev",
          "Name": "David",
          "id": 1,
          "objIndex": 0,
        },
      ],
      "filterDisabled": false,
      "rawData": undefined,
      "selectedFilters": Array [],
      "shouldFilterReset": false,
    }
  }
>
  <TableFilter
    applyFilter={[Function]}
    disabled={true}
    filterDisabled={false}
    filterableColumns={Array []}
    selectedFilters={Array []}
    setSelectedFilters={[Function]}
    shouldFilterReset={false}
  />
</ContextProvider>
`;
